Android Source/mobile/src/main/java/net/steppschuh/sensordatalogger/PhoneActivity.java

```diff
@@ -117,6 +129,54 @@ public void onClick(View v) {
             }
         });
 
+        floatingActionRecordButton = (FloatingActionButton) findViewById(R.id.recordActionButton);
+        floatingActionRecordButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // request runtime permission
+                if (ContextCompat.checkSelfPermission(getApplicationContext(),
+                        Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                        != PackageManager.PERMISSION_GRANTED) {
+                    if (ActivityCompat.shouldShowRequestPermissionRationale(app.getContextActivity(),
+                            Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                    } else {
+                        ActivityCompat.requestPermissions(app.getContextActivity(),
+                                new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE},23
```

Comment:
The `23` is the request code that you would get back in the result callback. Although the value itself doesn't matter too much, this looks like a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)). Usually you would define a constant for that, like `private static final int STORAGE_PERMISSION_REQUEST_CODE = 1;`

Reply:
(3/3) Removed the permission checking, good to know that it was not required. :+1: